### ISSUE TITLE
Ability to trigger a recipe cleanup programatively

### ISSUE THREAD

**Issue by vincentfalk** [ing-bank/baker#143]
There is currently no implementation allowing a **manual** cleanup of a recipe execution, like for example  

``
jbaker.cleanUp(processId)
``
or
``
jbaker.delete(processId)
``
or
``
jbaker.end(processId)
``

The possibility is to apply on the recipe
``
.withRetentionPeriod(Duration.ofDays(n))
``

In our use case of Baker, we would like to trigger ourselves the cleanup of the recipe executions that reached a  state that we consider as "final", at a given time scheduled in a batch cleanup mechanism and not based on an expiration period.

As discussed in a question with Merlijn, this could be done without too many issues as 

>This functionality is already there since it is used internally in the form of 'retention period'.

**Comment by merlijn** [comment]
Yes, it's a nice feature that I guess is sort of expected for a process orchestration a library.

One consideration is running interactions.

Maybe optionally stop them gracefully? For the retention period we currently force stop them.
